blockchain/v1/peer.go

```diff
@@ -46,13 +47,14 @@ func NewBpPeer(peerID p2p.ID, base int64, height int64,
 		params = BpPeerDefaultParams()
 	}
 	return &BpPeer{
-		ID:     peerID,
-		Base:   base,
-		Height: height,
-		blocks: make(map[int64]*types.Block, maxRequestsPerPeer),
-		logger: log.NewNopLogger(),
-		onErr:  onErr,
-		params: params,
+		ID:       peerID,
+		Base:     base,
+		Height:   height,
+		blocks:   make(map[int64]*types.Block, maxRequestsPerPeer),
+		noBlocks: make(map[int64]struct{}),
```

Comment:
Maybe we could just reuse `blocks` for this, by creating an explicit `nil` entry when there is no block?

Reply:
It's possible. We're already using it to record that request is sent https://github.com/tendermint/tendermint/blob/c2d37225545242539139dc7f2a163ccc5d6a91c6/blockchain/v1/peer.go#L151. Given we're going to remove blockchain/v1, I'd prefer explicit solution over reusing blocks.

Reply:
Ah right. Makes sense.